kaitai_utils.py

```diff
@@ -92,30 +118,35 @@ def _object_to_dict(self, instance: Any, path: str) -> Generator[Dict[str, Any],
         @param path: string representing the jsonpath to the current node in the object tree
         @return: dictionary containing parsed fields and their respective parsed values in a dictionary
         """
+        new_debug = None
         parameters_dict = _parameters_dict(instance)
+        if '_debug' in parameters_dict:
+            new_debug = parameters_dict['_debug']
         for key, value_object in parameters_dict.items():
-            if not _is_public_property(key, value_object):
-                continue
-            if _is_kaitai_struct(value_object):
-                yield _to_lower_camel_case(key), self._object_to_dict(value_object, path + '.' + key)
-            elif _is_list(value_object):
-                yield _to_lower_camel_case(key), self._list_to_dict(value_object, path)
-            elif isinstance(value_object, bytes):
-                yield self.process_byte_array(value_object, path, key)
-            else:
-                yield _to_lower_camel_case(key), _process_value(value_object)
-
-    def process_byte_array(self, value_object: bytes, path: str, key: str):
-        if len(value_object) > self.max_byte_array_length:
-            if len(value_object) < hansken_extraction_plugin.runtime.constants.MAX_CHUNK_SIZE:
-                child_builder = self.trace.child_builder(path)
-                child_builder.update(data={'raw': value_object}).build()
-                yield _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
-            else:
-                yield _to_lower_camel_case(
-                    key), f'data block of size: {len(value_object)} (not added as child trace because size exceeds MAX_CHUNK_SIZE)'
-        else:
-            yield _to_lower_camel_case(key), value_object.hex()
+            if _is_public_property(key, value_object):
+                if _is_kaitai_struct(value_object):
+                    yield _to_lower_camel_case(key), self._object_to_dict(value_object, path + '.' + key, new_debug)
+                elif _is_list(value_object):
+                    yield _to_lower_camel_case(key), self._list_to_dict(value_object, path)
+                elif isinstance(value_object, bytes):
+                    if len(value_object) > self.max_byte_array_length:
+                        if len(value_object) < hansken_extraction_plugin.runtime.constants.MAX_CHUNK_SIZE:
+                            length = new_debug[key]['end'] - new_debug[key]['start']
+                            child_builder = self.trace.child_builder(path)
+                            if self.has_process:
+                                child_builder.update(data={'raw': value_object}).build()
+                                yield _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
+                            else:
+                                yield _to_lower_camel_case(key), f'data block of size: {len(value_object)} (stored as {path})'
```

Comment:
Just a question: Should `yield` be on the first line of this `else` statement? When will the rest of the code be executed?

Reply:
That makes the code more readable indeed, but it does not change the resulting traces. Changed it anyway (Bouke gave me a nice `yield` explanation).